README.md

```diff
@@ -272,14 +276,17 @@ Over time, standardization of such signals is desired, though it may also be imp
-*   For nested resource loads, [a variant of the site for cookies algorithm](https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-rfc6265bis-10#section-5.2.1) should be used to avoid unrelated iframes from using a grant, with a permission policy opt-out, as suggested in.
-*   `SameSite=None` cookies granted via requestStorageAccessForOrigin on subresources should only be attached on CORS-enabled requests. For example, an `<img>` without a [`crossorigin` attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/crossorigin) would not have `SameSite=None` cookies attached, even with a valid grant. This ensures the server is aware of the caller and can react accordingly. Note that CORS is not possible on navigations; this requirement is not intended to protect `<iframe>` elements, which have other mechanisms for forbidding embedding (e.g., `x-frame-options` or `CSP`).
+*   For nested resource loads, [a variant of the site for cookies algorithm](https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-rfc6265bis-10#section-5.2.1) could be used to avoid unrelated iframes from using a grant, potentially with a permission policy opt-out, as suggested in [a recent security analysis](https://github.com/privacycg/storage-access/issues/113).
+  * Another alternative would be requiring explicit per-frame opt-in via normal `requestStorageAccess` call; see below.
+*   `SameSite=None` cookies granted via `requestStorageAccessForOrigin` on subresources should only be attached on CORS-enabled requests. For example, an `<img>` without a [`crossorigin` attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/crossorigin) set to `use-credentials` would not have `SameSite=None` cookies attached, even with a valid grant. This ensures the server is aware of the caller and can react accordingly; because the response must have the appropriate header to be read by the embedder, this ensures the embeddee has opted in. Note that CORS is not possible on navigations; this requirement is not intended to protect `<iframe>` elements.
+  * Subresource requests outside the top-level document that invoked `requestStorageAccessForOrigin` could also exclude cookies granted by the API (preventing unrelated frames from getting access).
+* For `<iframe>` elements, an explicit call to `requestStorageAccess` could be required before any cookies were made available. This would ensure opt-in by the embeddee, and align with the per-frame model of `requestStorageAccess`. The `requestStorageAccessForOrigin` call would still lift the `<iframe>` user interaction requirement and give loading control to the top-level document, and thus improve utility.
```

Comment:
Maybe instead of saying `would still lift the `<iframe>` user interaction requirement` we should explain how rSA would behave, i.e. `A requestStorageAccess call by a site with a requestStorageAccessForOrigin grant would behave as if a prior storage-access permission had been set for that site, i.e. resolve without requiring a user gesture or additional implementation-defined checks`.

Reply:
updated!